### PR TITLE
Logout

### PR DESCRIPTION
**Pull request by kamok**
__Description of Change(s) Introduced:__ 
Added a button on the top nav bar UI right of the Powered by Stride logo to log out users.

Added a new route called /logout

Added a controller action on AuthController called logout

__Description of Testing Applied:__ (if applicable)

__Relevant github Issue:__ (if applicable)

Issue #486

### CODE REVIEW

**Comment by kamok** [comment]
We should probably lock the yarn version as well. yarn.lock is creating a lot of noise since it was a -g install

**Comment by vanderhoop** [comment]
hey @kamok! thanks for submitting this. i'll find some time on your cal for friday so we can dig into the testing issues you were running into. also, please sign the cla agreement when you have a moment!

**Comment by kamok** [comment]
Will remove the e2e commit, I don't think it's necessary.

File: lib/remote_retro_web/templates/layout/app.html.eex [hunk 1]
```diff
@@ -70,13 +70,17 @@
...
-        <div class="ui right floated header right-branded-header">
+        <div class="ui right floated header right-branded-header grid">
           <a href="http://www.stridenyc.com/" target="_blank">
             <div>
               <div class="middle aligned content logo-text">Built by</div>
               <div class="ui tiny image"><img src="<%= static_path(@conn, "/images/stride_logo.png") %>" alt="Stride" /></div>
             </div>
           </a>
+
+          <a href="/logout" class="ui button mini">
```
Comment (vanderhoop) [inline lib/remote_retro_web/templates/layout/app.html.eex]: @kamok Realizing that we'll only want to display this button if there _is_ a current user, as it'll show up in the `/faq` page

File: test/controllers/auth_controller_test.exs [hunk 2]
```diff
@@ -70,6 +70,26 @@ defmodule RemoteRetro.AuthControllerTest do
     end
   end
 
+  describe ".logout" do
+    setup :authenticate_connection
+
+    test "it sets session current_user to nil", ~M{conn} do
```
Comment (vanderhoop) [inline test/controllers/auth_controller_test.exs]: 👍 💯

File: lib/remote_retro_web/templates/layout/app.html.eex (hunk 1 above)
Comment (vanderhoop) [reply lib/remote_retro_web/templates/layout/app.html.eex]: lemmeknow if you want to take that on, or if you want me to add it this weekend? happy to do it, as it was an oversight on my part

Comment (kamok) [reply lib/remote_retro_web/templates/layout/app.html.eex]: I can do it. Should be straightforward.

Comment (kamok) [reply lib/remote_retro_web/templates/layout/app.html.eex]: Doing it.

**Comment by kamok** [comment]
I implemented the show/hide of the log out button based on current_user in session. The exact implementation requires scrutiny to see if it matches best Phoenix practices. 

I have not written tests for it, as I think it requires me to write e2e, and e2e hasn't worked for me yet.

**Pull request merged**